Linked nodes validity improvements

sebeder opened the pull request:
Closes #247

sebeder commented:
Closes #248

wehrlean reviewed (changes requested):
If you recycle a model with a node, which is linked in another model. The error appearce correct but the reference is not deleted. It links to the deleted model.
![image](https://user-images.githubusercontent.com/49229000/107492282-8e2f7380-6b8c-11eb-925e-6ebeae18c14d.png)

Nodes in recycled models can be linked

If you recycle and the delete the model, the references of the nodes are not deleted. Only after reload.

sebeder commented:
Hi @wehrlean , I addressed points 2 and three. I decided, that if you recycle a model, its links are still maintained. However, if you hard delete it, the links are removed.
